flexget/plugins/notifiers/notify.py

```diff
@@ -0,0 +1,161 @@
+                     'minProperties': 1}]}},
+            'scope': {'type': 'string', 'enum': ['task', 'entries'], 'default': 'entries'},
+            'what': one_or_more({'type': 'string', 'enum': ENTRY_CONTAINERS}),
+            'title': {'type': 'string'},
+            'message': {'type': 'string'},
+            'url': {'type': 'string'},
+            'file_template': {'type': 'string'}
+        },
+        'required': ['to'],
+        'additionalProperties': True
+    }
+
+    def prepare_config(self, config):
+        config.setdefault('scope', 'entries')
+        config.setdefault('what', ['accepted'])
+        if not isinstance(config['what'], list):
+            config['what'] = [config['what']]
+        config.setdefault('title', DEFAULT_DICTS[config['scope']]['title'])
+        config.setdefault('message', DEFAULT_DICTS[config['scope']]['message'])
+        config.setdefault('url', DEFAULT_DICTS[config['scope']]['url'])
+        return config
+
+    @staticmethod
+    def render_value(entity, data, attribute, default_dict, plugin_name=None):
+        """
+        Tries to render a template, fallback to default template and just value if unsuccessful
+
+        :param entity: The entity to operate on, either `Entry` or `Task`
+        :param data: The text to be rendered
+        :param attribute: Attribute name to be fetched from the defaults
+        :param default_dict: The default dict, depending on entity type
+        :return: A rendered value or original value
+        """
+        result = data
+
+        # Handles file templates
+        if attribute == 'file_template':
+            try:
+                data = get_template(data, plugin_name)
+            except ValueError as e:
+                log.warning(e.args[0])
+                return
+
+        try:
+            result = entity.render(data)
+        except (RenderError, ValueError) as e:
+            log.debug('failed to render: %s. Trying to fall back to default', e.args[0])
+            try:
+                if attribute in default_dict:
+                    result = entity.render(default_dict[attribute])
+            # Render error on defaults should not happen
+            except RenderError as e:
+                log.warning('default dict failed to render: %s. Reverting to original value.', e.args[0])
+        else:
+            log.debug('successfully rendered `%s` to %s', attribute, result)
+        return result
+
+    def send_notification(self, task, config):
+        config = self.prepare_config(config)
+        scope = config.pop('scope')
+        what = config.pop('what')
+        notifiers = config.pop('to')
+
+        # Creating iterator based on config scope
+        iterate_on = [getattr(task, container) for container in what] if scope == 'entries' else [[task]]
+
+        for notifier in notifiers:
+            for notifier_name, notifier_config in notifier.items():
+                notifier = plugin.get_plugin_by_name(notifier_name).instance
```

Comment:
you might want to extract this into a method, or split this otherwise as it's long side